### PR TITLE
Update nextjs to version 14

### PR DESCRIPTION
**Pull request by gazzdingo**
### Features and Changes
updated the version of next to 14 so that we are able to support tailwind and more modern next features

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ If you just want to contribute a client library in a new language and not make c
 ## Requirements
 
 - MacOS or Linux (Windows may work too, but we haven't tested it)
-- [NodeJS](https://nodejs.org/en/download/package-manager/) 16.x or higher
+- [NodeJS](https://nodejs.org/en/download/package-manager/) 18.x
```
Comment (gazzdingo) [inline CONTRIBUTING.md:10]: because of microsoft azure we are unable to go above version 18 so i updated copy to state that it is version 18 only

**Comment by bttf** [comment]
Just poking around on the preview environment - the app loads and I'm able to sign in which is good. However when I tried creating an experiment I got this error, not sure if it's related or if it just happens in general on preview envs .. (I tried creating on another preview env and seemed to work fine.)
image

**Comment by gazzdingo** [comment]
> Just poking around on the preview environment - the app loads and I'm able to sign in which is good. However when I tried creating an experiment I got this error, not sure if it's related or if it just happens in general on preview envs .. (I tried creating on another preview env and seemed to work fine.) image

do you know if there is away to get the logs from the preview enviroment?

**Comment by bttf** [comment]
> > Just poking around on the preview environment - the app loads and I'm able to sign in which is good. However when I tried creating an experiment I got this error, not sure if it's related or if it just happens in general on preview envs .. (I tried creating on another preview env and seemed to work fine.) image
> 
> do you know if there is away to get the logs from the preview enviroment?

The logs are available in okteto for the backend instance and the mongo db. I tried looking for a way to invite but couldn't find one. Either way, logs don't show much ... seems like a frontend issue

image

**Review by bttf — COMMENTED** [review]
overall straightforward. just one question about `legacyBehavior`, don't think it's a blocker.

booted the site locally and clicked around - looks good. 

`yarn install` gave me a nasty non-blocking error. did you get that as well? might be nice to fix or silence if so.

File: packages/front-end/components/ActivityList.tsx [hunk 2]
```diff
@@ -41,53 +41,54 @@ const ActivityList: FC<{
...
+                  {/*        <strong>Reason: </strong> {details.reason}*/}
+                  {/*      </span>*/}
+                  {/*    )}*/}
+                  {/*    {details && details.results && (*/}
+                  {/*      <>*/}
+                  {/*        <strong>Result: </strong> {details.results}*/}
+                  {/*      </>*/}
+                  {/*    )}*/}
+                  {/*  </small>*/}
+                  {/*)}*/}
```
[the remark this answers is not in the document]
Comment (gazzdingo) [reply packages/front-end/components/ActivityList.tsx:90]: oh not sure why this is all commented out let me take a look

**Pull request closed without merging**